syntax = "proto3";

option java_package = "pasaud.voip.protocol";
package protocol;

import "google/protobuf/timestamp.proto";

message ClientBase {
  optional ClientAudio clientExtAudio = 1;
  optional Client clientExt = 2;
}

// [START messages]
message Client {
  optional uint32 secret_id = 1; // Secret ID to connect
  optional uint32 id = 2;  // Unique public ID number for this person.
  optional bytes audio = 3;
  optional uint32 audioNum = 4;
  optional float coordX = 5;
  optional float coordY = 6;
  optional float coordZ = 7;
  optional uint32 mapNum = 8;
  optional uint32 sampleTime = 9;

  optional google.protobuf.Timestamp packetTime = 10;
}

message ClientAudio {
  optional uint32 id = 1;
  optional bytes audio = 2;
  optional uint32 audioNum = 3;
  optional uint32 sampleTime = 4;
}


message Server {
  optional bool handShake = 1;
  uint32 id = 2;  // Unique public ID number for this person.
  optional bytes audio = 3;
  optional uint32 audioNum = 4;
  optional float coordX = 5;
  optional float coordY = 6;
  optional float coordZ = 7;
  optional uint32 mapNum = 8;
  optional uint32 sampleTime = 9;
  optional bool InvalidSession = 10;
  optional bool isGroup = 11;

  optional google.protobuf.Timestamp packetTime = 12;
  repeated bytes extraClientMSG = 13;
}